python_modules/dagster/dagster/_core/definitions/utils.py

```diff
@@ -118,16 +136,33 @@ def validate_tags(
         else:
             valid_tags[key] = value
 
+        if not is_valid_definition_tag_key(key):
+            invalid_tag_keys.append(key)
+
+    if invalid_tag_keys:
+        invalid_tag_keys_sample = invalid_tag_keys[: min(5, len(invalid_tag_keys))]
```

Comment:
limiting to 5 seems very arbitrary. I would just hard error and return all invalid tags